### PR TITLE
[Do not close or merge]  PowerUps Feature

### PR DESCRIPTION
**Pull request by seanlowjk**
Have Added the Following: 

1) `GYM_Visibility 1` For Playing of Power Ups 
2) Sample Prefabs (Currently Cubes) for Jump and Speed PowerUps 
3) Scripts from previous `feature/powerups` branch.

### CODE REVIEW

File: Assets/Scripts/PlayerController/PlayerMovement.cs [hunk 1]
```diff
@@ -28,6 +28,9 @@ public class PlayerMovement : MonoBehaviour
     private float m_JumpHeight = 1.3f;
 
     private CharacterController m_CharacterController;
+    private TextChanger m_textChanger;
+    private PowerUpsManager m_PowerUps;
```
Comment (Eclmist) [inline Assets/Scripts/PlayerController/PlayerMovement.cs]: PowerUpsManager sounds important. Consider making it a Singleton: https://wiki.unity3d.com/index.php/Singleton
I believe Caleb has already committed the Singleton script into `e3`, so you can make `PowerupsManager` inherit `Singleton`.

File: Assets/Scripts/PowerUps/VelocityModifier.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class VelocityModifier : MonoBehaviour
```
Comment (Eclmist) [inline Assets/Scripts/PowerUps/VelocityModifier.cs]: This script looks like it either requires `PlayerMovement` to function or `PlayerMovement` requires it to function. Consider adding the `[RequireComponent(typeOf()))]` tag in one of the classes.

File: Assets/Scripts/PlayerController/PlayerMovement.cs [hunk 3]
```diff
@@ -28,6 +28,9 @@ public class PlayerMovement : MonoBehaviour
     private float m_JumpHeight = 1.3f;
 
     private CharacterController m_CharacterController;
+    private TextChanger m_textChanger;
```
Comment (Eclmist) [inline Assets/Scripts/PlayerController/PlayerMovement.cs]: Why do you need text changer in here? Don't we already use this in `PowerupsManager`?

Comment (seanlowjk) [reply Assets/Scripts/PlayerController/PlayerMovement.cs]: Yikes didn't notice! Will fix in next commit!

File: Assets/Scripts/PowerUps/VelocityModifier.cs (hunk 2 above)
Comment (seanlowjk) [reply Assets/Scripts/PowerUps/VelocityModifier.cs]: Okay sures!

File: Assets/Scripts/PlayerController/PlayerMovement.cs (hunk 1 above)
Comment (seanlowjk) [reply Assets/Scripts/PlayerController/PlayerMovement.cs]: Sick stuff, kk will fetch then make the changes

Comment (seanlowjk) [reply Assets/Scripts/PlayerController/PlayerMovement.cs]: Actually on second thoughts, I am unable to make it a singleton for now as it requires different boolean values on the state of the power ups of each player. 

As each player requires their own instance of PowerUps Manager. As a result, we would need minimally 4. For now, I will not put it as a Singleton.

**Pull request closed without merging**